Applications/Custom/OptimizerClient/main.cpp

```diff
@@ -29,5 +68,71 @@ int main() {
     return 1;
   }
 
+  auto model =
+    ml::train::createModel(ml::train::ModelType::NEURAL_NET, {"batch_size=10"});
+
+  std::shared_ptr<ml::train::Dataset> dataset;
+  std::shared_ptr<ml::train::Optimizer> optimizer;
+  std::vector<std::shared_ptr<ml::train::Layer>> layers;
+
+  try {
+    dataset = ml::train::createDataset(ml::train::DatasetType::GENERATOR,
+                                       constant_generator_cb);
+  } catch (...) {
+    std::cerr << "creating dataset failed";
+    return 1;
+  }
+
+  if (model->setDataset(ml::train::DatasetModeType::MODE_TRAIN, dataset) != 0) {
```

Comment:
this can also throw?

Reply:
We should make this void to avoid double error checking :p, updated